### PR TITLE
TextExtractor-20388: add SupportedLanguages and Troubleshooting sections

### PR DESCRIPTION
**Pull request by taras-janea**

### CODE REVIEW

**Comment by taras-janea** [comment]
cc: @jaimecbernardo

File: hub/powertoys/text-extractor.md [hunk 1]
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
...
+```
+and these remove:
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Remove-WindowsCapability -Online
+```
+
+## Troubleshooting
+
+* "No Possible OCR languages are installed." message </br>
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: Please use `-` instead of `*` as in other pages.

File: hub/powertoys/text-extractor.md [hunk 2]
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
 | Setting | Description |
 | :--- | :--- |
 | Activation shortcut | The customizable keyboard command to turn on or off always on top for that window. |
+
+## Supported languages
+
+Text Extractor can recognize only languages with OCR pack available.</br>
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: Text Extractor can only recognize languages that have the OCR pack installed.

File: hub/powertoys/text-extractor.md [hunk 3]
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
...
+The list can be obtained via PowerShell by running next commands:
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: "next commands" -> "following commands"

File: hub/powertoys/text-extractor.md [hunk 4]
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
...
+```console
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
+```
+
+### OCR packs installation
+
+If language is supported, then OCR language pack is installed along with language via <kbd>Language</kbd>-><kbd>Preferred languages</kbd>-><kbd>Add language menu</kbd> by default.</br>
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: "For supported languages, the OCR language pack can be installed along with the language via"
And remove the "by default."

File: hub/powertoys/text-extractor.md [hunk 5]
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
...
+
+It's possible to query all supported languages via PowerShell run as Administrator:
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: "It's possible to get a list of all supported languages by running this command in PowerShell in Administrator mode:"

File: hub/powertoys/text-extractor.md [hunk 6]
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
...
+
+Name  : Language.OCR~~~es-ES~0.0.1.0
+State : NotPresent
+
+Name  : Language.OCR~~~es-MX~0.0.1.0
+State : NotPresent
+....
+```
+
+If language is not available in the output, then it's not supported by OCR.</br>
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: "If a language"

File: hub/powertoys/text-extractor.md [hunk 7]
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
...
+If the state is <kbd>NotPresent</kbd>, then it's possible to install OCR pack for the language.</br>
+For example, the next commands install OCR pack for <kbd>en-US</kbd> in PowerShell run as Administrator:
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: "install OCR pack" -> "install the OCR pack"

**Review by jaimecbernardo — CHANGES_REQUESTED** [review]
I think this needs some language changes.
Instead of using <br> blocks, please use a blank line between sentences as it's done in other documentation pages.
Other than that, it LGTM!

**Comment by jaimecbernardo** [comment]
Please add @crutkas to reviewers as well.

File: hub/powertoys/text-extractor.md [hunk 8]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+If a language is not available in the output, then it's not supported by OCR.
+
+If the state is <kbd>NotPresent</kbd>, then it's possible to install OCR pack for the language.
+
+For example, the following commands install the OCR pack for <kbd>en-US</kbd> in PowerShell run as Administrator:
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
+and these remove:
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: These commands remove it:

File: hub/powertoys/text-extractor.md [hunk 9]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Remove-WindowsCapability -Online
+```
+
+## Troubleshooting
+
+- "No Possible OCR languages are installed." message
+
+Message is shown, when there are no available languages for recognition.
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: "This message is shown when there are no available languages for recognition."

File: hub/powertoys/text-extractor.md [hunk 10]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+
+If OCR pack is supported, installed, but still not available and your system drive <kbd>X:</kbd> is different than <kbd>C:</kbd>, then copy <kbd>X:/Windows/OCR</kbd> folder to <kbd>C:/Windows/OCR</kbd> to fix the issue.
```
Comment (jaimecbernardo) [inline hub/powertoys/text-extractor.md]: "If an OCR pack"

**Review by jaimecbernardo — APPROVED** [review]
Some nits, other than that LGTM! Thank you!

File: hub/powertoys/text-extractor.md [hunk 11]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
+These commands remove it:
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Remove-WindowsCapability -Online
+```
+
+## Troubleshooting
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md:107]: how does Install a language pack needs to be a section

Comment (taras-janea) [reply hub/powertoys/text-extractor.md:107]: Have removed "### OCR packs installation" line. I guess that's what you meant, right?

Comment (crutkas) [reply hub/powertoys/text-extractor.md:107]: no, if i don't have the OCR pack needed, we should say how they get it.  "Hey, i want german, ..."

File: hub/powertoys/text-extractor.md [hunk 12]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+If a language is not available in the output, then it's not supported by OCR.
+
+If the state is <kbd>NotPresent</kbd>, then it's possible to install OCR pack for the language.
+
+For example, the following commands install the OCR pack for <kbd>en-US</kbd> in PowerShell run as Administrator:
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
+These commands remove it:
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
#### How to remove a language pack
```

File: hub/powertoys/text-extractor.md [hunk 13]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+State : NotPresent
+
+Name  : Language.OCR~~~es-MX~0.0.1.0
+State : NotPresent
+....
+```
+
+If a language is not available in the output, then it's not supported by OCR.
+
+If the state is <kbd>NotPresent</kbd>, then it's possible to install OCR pack for the language.
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
#### How to install a language pack
```

File: hub/powertoys/text-extractor.md [hunk 14]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+```console
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
+```
+
+### OCR packs installation
+
+For supported languages, the OCR language pack can be installed along with the language via <kbd>Language</kbd>-><kbd>Preferred languages</kbd>-><kbd>Add language menu</kbd>.
+
+It's possible to get a list of all supported languages by running this command in PowerShell in Administrator mode:
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
#### How to query for language packs
```

**Review by crutkas — CHANGES_REQUESTED** [review]

File: hub/powertoys/text-extractor.md [hunk 15]
```diff
@@ -14,6 +14,12 @@ Text Extractor is a convenient way to copy text from anywhere on your screen. Th
...
 
+## How to deactivate
+
+Capture mode is deactivated immediately after text in the selected region is recognized and copied to the clipboard.
+
+You can exit capture mode by pressing <kbd>Esc</kbd> at any moment.
+
 ## Adjust while trying to capture
 
 By holding <kbd>shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>shift</kbd>, you will be able to resize again.
```
Comment (Jay-o-Way) [inline hub/powertoys/text-extractor.md]:
```suggestion
By holding <kbd>Shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>Shift</kbd>, you will be able to resize again.
```
Caps

File: hub/powertoys/text-extractor.md [hunk 16]
```diff
@@ -14,6 +14,12 @@ Text Extractor is a convenient way to copy text from anywhere on your screen. Th
 
 With the activation shortcut (default: <kbd>⊞ Win</kbd>+<kbd>Shift</kbd>+<kbd>T</kbd>), you'll see an overlay on the screen. Click and hold your primary mouse button and drag to activate your capture. The text will be saved to your clipboard.
 
+## How to deactivate
+
+Capture mode is deactivated immediately after text in the selected region is recognized and copied to the clipboard.
+
+You can exit capture mode by pressing <kbd>Esc</kbd> at any moment.
```
Comment (Jay-o-Way) [inline hub/powertoys/text-extractor.md]:
```suggestion
Capture mode is deactivated immediately after text in the selected region is recognized and copied to the clipboard. You can exit capture mode by pressing <kbd>Esc</kbd> at any moment.
```
Why new paragraph?

File: hub/powertoys/text-extractor.md [hunk 17]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+PS C:\Windows\system32> $Capability | Remove-WindowsCapability -Online
+```
+
+## Troubleshooting
+
+- "No Possible OCR languages are installed." message
+
+This message is shown when there are no available languages for recognition.
+
+If an OCR pack is supported and installed, but still is not available and your system drive <kbd>X:</kbd> is different than <kbd>C:</kbd>, then copy <kbd>X:/Windows/OCR</kbd> folder to <kbd>C:/Windows/OCR</kbd> to fix the issue.
```
Comment (Jay-o-Way) [inline hub/powertoys/text-extractor.md]:
```suggestion
If an OCR pack is supported and installed, but still is not available and your system drive _X:_ is different than "C:", then copy `X:/Windows/OCR`  folder to `C:/Windows/OCR` to fix the issue.
```
These aren't keys on a keyboard.

File: hub/powertoys/text-extractor.md [hunk 18]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+
+The following commands install the OCR pack for <kbd>en-US</kbd>:
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
+
+### How to remove an OCR language pack
+
+The following commands remove the OCR pack for <kbd>en-US</kbd>:
```
Comment (Jay-o-Way) [inline hub/powertoys/text-extractor.md]:
```suggestion
The following commands remove the OCR pack for "en-US":
```

File: hub/powertoys/text-extractor.md [hunk 19]
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
...
+Name  : Language.OCR~~~es-MX~0.0.1.0
+State : NotPresent
+....
+```
+
+If a language is not available in the output, then it's not supported by OCR.
+
+### How to install an OCR language pack
+
+The following commands install the OCR pack for <kbd>en-US</kbd>:
```
Comment (Jay-o-Way) [inline hub/powertoys/text-extractor.md]:
```suggestion
The following commands install the OCR pack for "en-US":
```

**Review by Jay-o-Way — CHANGES_REQUESTED** [review]
kbd tags & syntax

File: hub/powertoys/text-extractor.md [hunk 20]
```diff
@@ -14,9 +14,14 @@ Text Extractor is a convenient way to copy text from anywhere on your screen. Th
...
+
+Capture mode is deactivated immediately after text in the selected region is recognized and copied to the clipboard.
+You can exit capture mode by pressing <kbd>Esc</kbd> at any moment.
+
 ## Adjust while trying to capture
 
-By holding <kbd>shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>shift</kbd>, you will be able to resize again.
+By holding <kbd>Shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>Shift</kbd>, you will be able to resize again.
 
 ## Things to note
```
Comment (Jay-o-Way) [inline hub/powertoys/text-extractor.md]: You could also place this in a special `Note` box, see https://learn.microsoft.com/contribute/markdown-reference#alerts-note-tip-important-caution-warning

**Review by Jay-o-Way — COMMENTED** [review]
Line 28 _could_ be simplified: "1. This tool uses Optical Character Recognition to read the text on the screen."

File: hub/powertoys/text-extractor.md (hunk 20 above)
Comment (taras-janea) [reply hub/powertoys/text-extractor.md]: I'd prefer to leave it as it is.

**Comment by taras-janea** [comment]
> Line 28 _could_ be simplified: "1. This tool uses Optical Character Recognition to read the text on the screen."

I'd prefer not to change it.

**Comment by Court72** [comment]
@mattwojo

Can you review the proposed changes? *IMPORTANT*: When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

File: hub/powertoys/text-extractor.md (hunk 11 above)
Comment (taras-janea) [reply hub/powertoys/text-extractor.md:107]: Hi Clint,

I've made changes according to your comments.
Could you please check whether it looks good to you?

Thank you,
Taras

File: hub/powertoys/text-extractor.md [hunk 21]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
+```console
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
+```
+### How to query for OCR language packs
+
+The next command returns the list (PowerShell run as Administrator):
+```console
+PS C:\Windows\system32> Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*' }
+```
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
```powershell
Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*' }
```
```

File: hub/powertoys/text-extractor.md [hunk 22]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
+
+## Supported languages
+
+Text Extractor can only recognize languages that have the OCR language pack installed.
+
+The list can be obtained via PowerShell by running the following commands:
+```console
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
+```
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
```powershell
[Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
[Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
```
```

File: hub/powertoys/text-extractor.md [hunk 23]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
+
+If a language is not available in the output, then it's not supported by OCR.
+
+### How to install an OCR language pack
+
+The following commands install the OCR pack for "en-US":
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
```powershell
$Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
$Capability | Add-WindowsCapability -Online
```
```

File: hub/powertoys/text-extractor.md [hunk 24]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
+
+### How to remove an OCR language pack
+
+The following commands remove the OCR pack for "en-US":
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Remove-WindowsCapability -Online
+```
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
```powershell
$Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
$Capability | Remove-WindowsCapability -Online
```
```

File: hub/powertoys/text-extractor.md [hunk 25]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
+
+## Troubleshooting
+
+- "No Possible OCR languages are installed." message
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]:
```suggestion
### "No Possible OCR languages are installed." message
```

**Review by crutkas — COMMENTED** [review]
tweak to my suggestion and approved

**Comment by taras-janea** [comment]
> tweak to my suggestion and approved

Thank you for your suggestions. Have applied all of them.

File: hub/powertoys/text-extractor.md (hunk 22 above)
Comment (crutkas) [reply hub/powertoys/text-extractor.md]: My change was larger. Having the c: blah blah blocks a copy paste

File: hub/powertoys/text-extractor.md [hunk 26]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
 | :--- | :--- |
 | Activation shortcut | The customizable keyboard command to turn on or off this module. |
+
+## Supported languages
+
+Text Extractor can only recognize languages that have the OCR language pack installed.
+
+The list can be obtained via PowerShell by running the following commands:
+```powershell
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]: The c: stuff needs to be removed. My suggestion had this removal included.

File: hub/powertoys/text-extractor.md [hunk 27]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
+```
+### How to query for OCR language packs
+
+The next command returns the list (PowerShell run as Administrator):
+```powershell
+PS C:\Windows\system32> Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*' }
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]: The c: stuff needs to be removed. My suggestion had this removal included.

File: hub/powertoys/text-extractor.md [hunk 28]
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
...
+....
+```
+
+If a language is not available in the output, then it's not supported by OCR.
+
+### How to install an OCR language pack
+
+The following commands install the OCR pack for "en-US":
+```powershell
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
```
Comment (crutkas) [inline hub/powertoys/text-extractor.md]: The c: stuff needs to be removed. My suggestion had this removal included.

**Review by crutkas — CHANGES_REQUESTED** [review]

File: hub/powertoys/text-extractor.md (hunk 22 above)
Comment (taras-janea) [reply hub/powertoys/text-extractor.md]: Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

File: hub/powertoys/text-extractor.md (hunk 26 above)
Comment (taras-janea) [reply hub/powertoys/text-extractor.md]: Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

File: hub/powertoys/text-extractor.md (hunk 27 above)
Comment (taras-janea) [reply hub/powertoys/text-extractor.md]: Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

File: hub/powertoys/text-extractor.md (hunk 28 above)
Comment (taras-janea) [reply hub/powertoys/text-extractor.md]: Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

**Review by crutkas — APPROVED** [review]

**Comment by crutkas** [comment]
@mattwojo should be good pending your sign off

File: hub/powertoys/text-extractor.md (hunk 11 above)
Comment (mattwojo) [reply hub/powertoys/text-extractor.md:107]: Looks good! Thank you Taras!

**Comment by mattwojo** [comment]
@crutkas - I tweaked the intro statement a bit to add a bit about grabbing text from an image or video (to provide more context for most common use-cases) and removed the "make it more convenient" statement as we try to avoid these sort of subjective statements like "it's easy". 

At some point it may be worth also doing an accessibility review of the PowerToys docs... there are a lot of instances of instruction for using a mouse and right-clicking (for example), this may be unavoidable, but we are generally trying to write for folks who may be using assistive technology as well (like alternatives to a mouse -- like a trackpad or... IDK!) ...just a thought for future discussion.

If that intro looks ok, please just add #sign-off to get this merged live. :)

**Review by mattwojo — APPROVED** [review]

**Pull request merged**